[Doc] Revamp ray core design patterns doc [12/n]: global variables (#29124)
ray-project/ray: doc/source/ray-core/patterns/global-variables.rst

> Anti-pattern: Accessing or modifying global variables in tasks and actors

Author: @stephanie-wang: do you have suggestions to the title to make it more like others: doing something causes/harms xxx

Reviewer: Hmm I guess we don't have a good "consequence" since it's really just unintuitive effect compared to single-process Python.

But how about updating to be more specific?
"Using global variables to share state between tasks and actors"

=========================================================================

> **TLDR:** Don't access or modify global variables in remote functions and classes. Instead, encapsulate the global variables into an actor's instance variables.

Reviewer: ```suggestion
**TLDR:** Don't use global variables to share state with tasks and actors. Instead, encapsulate the global variables in an actor and pass the actor handle to other tasks and actors.
```

> Ray drivers, tasks and actors are running in
> different processes that don’t share the same address space with each other.
> That says if you modify global variables
> in one process, changes are not reflected in other processes.

Reviewer: ```suggestion
Ray drivers, tasks and actors are running in
different processes, so they don’t share the same address space.
This means that if you modify global variables
in one process, changes are not reflected in other processes.
```
